go-controller/pkg/node/secondary_node_network_controller_test.go

```diff
@@ -156,4 +218,69 @@ var _ = Describe("SecondaryNodeNetworkController", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(controller.gateway).To(BeNil())
 	})
+	It("ensure UDNGateway and VRFManager is invoked for Primary UDNs when feature gate is ON", func() {
+		config.OVNKubernetesFeature.EnableNetworkSegmentation = true
+		config.OVNKubernetesFeature.EnableMultiNetwork = true
+		factoryMock := factoryMocks.NodeWatchFactory{}
+		node := &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName,
+				Annotations: map[string]string{
+					"k8s.ovn.org/network-ids":  fmt.Sprintf("{\"%s\": \"%d\"}", netName, netID),
+					"k8s.ovn.org/node-subnets": fmt.Sprintf("{\"%s\":[\"%s\", \"%s\"]}", netName, v4NodeSubnet, v6NodeSubnet)},
+			},
+		}
+		nodeList := []*corev1.Node{node}
+		cnnci := CommonNodeNetworkControllerInfo{name: nodeName, watchFactory: &factoryMock}
+		factoryMock.On("GetNode", nodeName).Return(nodeList[0], nil)
+		factoryMock.On("GetNodes").Return(nodeList, nil)
+		nodeInformer := coreinformermocks.NodeInformer{}
+		factoryMock.On("NodeCoreInformer").Return(&nodeInformer)
+		nodeLister := v1mocks.NodeLister{}
+		nodeInformer.On("Lister").Return(&nodeLister)
+		nodeLister.On("Get", mock.AnythingOfType("string")).Return(node, nil)
+		cnode := node.DeepCopy()
+		cnode.Annotations[util.OvnNodeManagementPortMacAddresses] = `{"bluenet":"00:00:00:55:66:77"}`
+		kubeMock.On("UpdateNodeStatus", cnode).Return(nil)
+		nad = ovntest.GenerateNAD("bluenet", "rednad", "greenamespace",
+			types.Layer3Topology, "100.128.0.0/16", types.NetworkRolePrimary)
+		NetInfo, err := util.ParseNADInfo(nad)
+		Expect(err).NotTo(HaveOccurred())
+		controller, err := NewSecondaryNodeNetworkController(&cnnci, NetInfo, vrf)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(controller.gateway).To(Not(BeNil()))
+		controller.gateway.kubeInterface = &kubeMock
+		err = testNS.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+			getCreationFakeOVSCommands(fexec, mgtPort, mgtPortMAC, netName, nodeName, NetInfo.MTU())
+			Expect(err).NotTo(HaveOccurred())
+			getDeletionFakeOVSCommands(fexec, mgtPort)
+			Expect(err).NotTo(HaveOccurred())
+			err = controller.Start(context.Background())
+			Expect(err).NotTo(HaveOccurred())
+			vrfDeviceName := util.GetVrfDeviceNameForUDN(netID)
+			vrfLink, err := util.GetNetLinkOps().LinkByName(vrfDeviceName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(vrfLink.Type()).To(Equal("vrf"))
+			vrfDev, ok := vrfLink.(*netlink.Vrf)
+			Expect(ok).To(Equal(true))
```

Comment:
can you add some ginkgo.by steps in here to communicate what you are testing and checking